tensorflow_probability/python/experimental/bijectors/highway_flow.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import tensorflow.compat.v2 as tf
+
+from tensorflow_probability.python import bijectors as tfb
+from tensorflow_probability.python import util
+from tensorflow_probability.python.experimental.bijectors import scalar_function_with_inferred_inverse
+from tensorflow_probability.python.internal import samplers
+
+
+def build_highway_flow_layer(width, residual_fraction_initial_value=0.5, activation_fn=None, seed=None):
+    # TODO: add control that residual_fraction_initial_value is between 0 and 1
+    residual_fraction_initial_value = tf.convert_to_tensor(residual_fraction_initial_value,
+                                                           dtype_hint=tf.float32,
+                                                           name='residual_fraction_initial_value')
+    dtype = residual_fraction_initial_value.dtype
+
+    bias_seed, upper_seed, lower_seed, diagonal_seed = samplers.split_seed(seed, n=4)
+
+    return HighwayFlow(
+        width=width,
+        residual_fraction=util.TransformedVariable(
+            initial_value=np.asarray(residual_fraction_initial_value),
+            bijector=tfb.Sigmoid(),
+            dtype=dtype),
+        activation_fn=activation_fn,
+        bias=tf.Variable(samplers.normal((width,), 0., 0.01, seed=bias_seed), dtype=dtype),
+        upper_diagonal_weights_matrix=util.TransformedVariable(
+            initial_value=np.tril(samplers.normal((width, width), 0., 1., seed=upper_seed), -1) + np.diag(
+                samplers.uniform((width,), minval=0., maxval=1., seed=diagonal_seed)),
+            bijector=tfb.FillScaleTriL(diag_bijector=tfb.Softplus(), diag_shift=None),
+            dtype=dtype),
+        lower_diagonal_weights_matrix=util.TransformedVariable(
+            initial_value=samplers.normal((width, width), 0., 1., seed=lower_seed),
+            bijector=tfb.Chain([tfb.TransformDiagonal(diag_bijector=tfb.Shift(1.)),
+                                tfb.Pad(paddings=[(1, 0), (0, 1)]),
+                                tfb.FillTriangular()]),
+            dtype=dtype)
+    )
+
+
+class HighwayFlow(tfb.Bijector):
```

Comment:
I think I figured out how FFJORD manages to cache the fldj/ildj computations from `_augmented_forward` / `_augmented_inverse`. You'll need to add a line to the HighwayFlow class similar to this one from FFJORD:

```
_cache = cache_util.BijectorCacheWithGreedyAttrs(
      forward_name='_augmented_forward',
      inverse_name='_augmented_inverse')
```
(with a comment similar to what's in https://github.com/tensorflow/probability/blob/master/tensorflow_probability/python/bijectors/ffjord.py#L246-L252)

Internally, calling `bijector.forward()` invokes `bijector._cache.forward()`, so this line sets up a cache that knows to call `_augmented_forward` and remember both of the results that it returns.